app/client/ui/selectBy.ts

```diff
@@ -224,15 +255,29 @@ function fromViewSectionRec(section: ViewSectionRec): LinkNode[] {
     return [];
   }
   const table = section.table.peek();
-  const ancestors = new Set<number>();
+  const ancestors: number[] = [];
+
+  const isAncestorSameTableCursorLink: boolean[] = [];
 
   for (let sec = section; sec.getRowId(); sec = sec.linkSrcSection.peek()) {
-    if (ancestors.has(sec.getRowId())) {
-      // tslint:disable-next-line:no-console
-      console.warn(`Links should not create a cycle - section ids: ${Array.from(ancestors)}`);
+    if (ancestors.includes(sec.getRowId())) {
+      // There's a cycle in the existing link graph
+      // TODO if we're feeling fancy, can test here whether it's an all-same-table cycle and warn if not
+      //      but there's other places we check for that
       break;
     }
-    ancestors.add(sec.getRowId());
+    ancestors.push(sec.getRowId());
+
+    //Determine if this link is a same-table cursor link
+    if(sec.linkSrcSection.peek().getRowId()) {
```

Comment:
```suggestion
    if (sec.linkSrcSection.peek().getRowId()) {
```

Why do we need this condition? Isn't it already checked in the for loop?

Reply:
We're one step ahead of the loop, the loop test for sec.linkSrcSection won't run until we're finished with this iteration, and I believe we end up going 1 past the end of the loop anyway, so we'll have to make some extra checks